test/usecases/solve/test_scalar.py

```diff
@@ -0,0 +1,33 @@
+from typing import Optional
+
+import numpy as np
+from neuron import h, gui
+from neuron.units import ms
+
+
+def simulate(method: str, rtol: float, dt: Optional[float] = None):
+    nseg = 1
+
+    s = h.Section()
+    s.insert(f"{method}_scalar")
+    s.nseg = nseg
+
+    x_hoc = h.Vector().record(getattr(s(0.5), f"_ref_x_{method}_scalar"))
+    t_hoc = h.Vector().record(h._ref_t)
+
+    h.stdinit()
+    if dt is not None:
+        h.dt = dt * ms
+    h.tstop = 5.0 * ms
+    h.run()
+
+    x = np.array(x_hoc.as_numpy())
+    t = np.array(t_hoc.as_numpy())
+
+    x_exact = 42.0 * np.exp(-t)
+    np.testing.assert_allclose(x, x_exact, rtol=rtol)
+
+
+if __name__ == "__main__":
+    simulate("cnexp", rtol=1e-12)
+    simulate("derivimplicit", rtol=1e-3, dt=1e-4)
```

Comment:
How large is the error with larger timesteps? Does it differ between NOCMODL and NMODL generated code?

Reply:
Good question! From my experiments, both NMODL and NOCMODL produce identical relative differences w.r.t. the exact result, with the following max relative error:
- `0.06340655` for default value of `h.dt`
- `0.00250146` for `h.dt = 1e-3 * ms`
- `0.00025001` for `h.dt = 1e-4 * ms`
- `2.49963513e-06` for `h.dt = 1e-6 * ms`

and so on.

Reply:
Perfect, exactly the convergence rate we want to see.